frame/protocols/__init__.py

```diff
@@ -0,0 +1,17 @@
+from pathlib import Path
+import pywayland
+import pywayland.scanner
+
+def generate_protocol(name, imports: dict[str, str]) -> dict[str, str]:
+    input_path = Path(__file__).parent.parent / 'data' / f'{name}.xml'
+    proto = pywayland.scanner.Protocol.parse_file(str(input_path))
+    proto_imports = {iface.name: proto.name for iface in proto.interface}
+    output_dir = Path(__file__).parent.parent / 'protocols'
```

Comment:
Edited my suggestion too late ;)

```suggestion
    input_path = Path(__file__).parents[1] / 'data' / f'{name}.xml'
    proto = pywayland.scanner.Protocol.parse_file(str(input_path))
    proto_imports = {iface.name: proto.name for iface in proto.interface}
    output_dir = Path(__file__).parents[1] / 'protocols'
```